File: Stack/Implement a Stack/stack.js
export default class Stack {
  constructor() {
    this._items = [];
  }

  push(item) {
    return this._items.push(item);
  }

  pop() {
    return this._items.pop();
  }

  isEmpty() {
    return this._items.length() === 0;
  }

  peek() {
    return this.isEmpty() ? undefined : this._items[this.length() - 1];
  }

  length() {
    return this._items.length;
  }
}

File: Stack/Implement a Stack/stack.py
class Stack:
    def __init__(self):
        self._items = []
    
    def push(self, item):
        self._items.append(item)
    
    def pop(self):
        return self._items.pop() if not self.is_empty() else None
    
    def is_empty(self):
        return self.length() == 0
  
    def peek(self):
        return self._items[-1] if not self.is_empty() else None

    def length(self):
        return len(self._items)
